### PR TITLE
[Verifier/Meter] Transfer module meter to package meter

### PR DESCRIPTION
**Pull request by amnn**
## Description

The motivation for this change is to properly track the modules being visited in the `Meter`, so that in our command line tool, we can provide an output that correctly tracks how many meter ticks were used to verify each function and module.

Prior to this change, the metered verifier pretended like there was only "one module" that all the functions being verified were in. This change introduces the level of packages that ticks from modules are transferred to.

We also add an explicit protocol config parameter to set a max tick limit per package (whose value is the same as the function and module level limits currently). This was not strictly necessary but was done to avoid confusion (caused by treating the module level limit as the package level limit).

This should be a behaviour preserving change, but in some cases we may do more work verifying transactions because of how the ticks for each module are now gathered in isolation before being transferred to the package scope. In any case, a behaviour change would be okay, because metering (and related errors) is only used during signing, and not execution.

## Test Plan

```
sui$ cargo simtest
deepbook$ cargo run --bin sui -p sui -- client verify-bytecode-meter
```

## Stack

- #16903 
- #16941

### CODE REVIEW

File: sui-execution/latest/sui-adapter/src/adapter.rs [hunk 1]
```diff
@@ -204,6 +183,42 @@ mod checked {
...
+        } else if let Err(err) = sui_verify_module_metered_check_timeout_only(
+            module,
+            &BTreeMap::new(),
+            meter,
+            verifier_config,
+        ) {
+            return Err(err.into());
+        }
+
+        if meter.transfer(Scope::Module, Scope::Package, 1.0).is_err() {
```
Comment (tnowacki) [inline sui-execution/latest/sui-adapter/src/adapter.rs:216]: This is the spot where we might do more work, right? 

And to check my understanding, the amount of additional work is bounded by the amount of work permissible for a given module?

Comment (amnn) [reply sui-execution/latest/sui-adapter/src/adapter.rs:216]: That's right, so we may spend as much as 2x the current tick limit before we realise we've expended the limit with this change. I think we could potentially be cleverer with the meter implementation though, to have the package donate its limit (up to the module limit) to the module when you enter it, and have the module return the unused limit when it comes back.

**Review by tnowacki — APPROVED** [review]

File: sui-execution/latest/sui-adapter/src/adapter.rs (hunk 1 above)
Comment (amnn) [reply sui-execution/latest/sui-adapter/src/adapter.rs:216]: This ...didn't work for a couple of reasons:

- Originally, I tried what I mentioned above, but it gets very messy because we don't keep track of where we are entering a scope from (nothing in the interface preventing us from entering a package from within a module, etc), so there isn't a good way to know which scope is donating to the one being entered.
- Then I tried keeping a global pool of ticks that every scope pulls from (and donates back to on `transfer`). This also doesn't work, because of the scale factor parameter on `transfer`.

TL;DR: The `Meter` trait confounds me once again.
